### ISSUE TITLE
[Bug]: fig.canvas.mpl_connect in 3.5.2 not registering events in jupyter lab unless using widget pan or zoom controls

### ISSUE THREAD

**Issue by StevenSong** [matplotlib/matplotlib#23470]
### Bug summary

In jupyter lab, the code snippet below sets the title of the axis to the x,y coords of a mouse click, however in matplotlib 3.5.2 specifically, the clicks aren't registered unless I'm clicking around using the pan or zoom controls of the interactive widget. Downgrading to matplotlib 3.5.1, the clicks are registered with just the cursor as expected with the same ipympl version 0.9.1.

### Code for reproduction

```python
import numpy as np
import matplotlib.pyplot as plt
%matplotlib widget

fig, ax = plt.subplots(figsize=(5,5))
plt.plot(np.random.randint(0, 5, 5))
plt.show()

def onclick(event):
    ax.set_title(f'x={event.x}, y={event.y}')
    print(f'x={event.x}, y={event.y}')

fig.canvas.mpl_connect('button_press_event', onclick)
```

### Actual outcome

No axis title after clicking inside the plot with just a regular cursor
<img width="514" alt="Screen Shot 2022-07-22 at 9 13 29 AM" src="https://user-images.githubusercontent.com/26208374/180457943-08bf7053-0870-45bc-88c7-f22791d8d26a.png">

Axis title set after clicking inside the plot with the pan tool of the interactive widget (same if using the zoom tool)
<img width="518" alt="Screen Shot 2022-07-22 at 9 12 03 AM" src="https://user-images.githubusercontent.com/26208374/180457956-538d1a22-c741-483f-881e-94f710761529.png">

### Expected outcome

This screenshot was taken with mpl version 3.5.1 where the regular cursor click was registered and the axis title was changed
<img width="519" alt="Screen Shot 2022-07-22 at 9 17 51 AM" src="https://user-images.githubusercontent.com/26208374/180458756-8db913fe-addc-428f-911b-d8232fa05133.png">

### Additional information

It's also interesting that under mpl 3.5.2, the cursor remains as the finger cursor icon after clicking while in mpl 3.5.1, the cursor icon changes back to the regular pointer - I'm not sure what that means but it is a difference in behavior...

Some relevant package versions:
matplotlib 3.5.2
ipympl 0.9.1
jupyterlab 3.4.3

### Operating system

reproduced on both Ubuntu and macOS

### Matplotlib Version

3.5.2

### Matplotlib Backend

module://ipympl.backend_nbagg

### Python version

3.8.13

### Jupyter version

3.4.3

### Installation

pip

**Comment by tacaswell** [comment]
I can reproduce this on v3.5.2 but can not reproduce it on the v3.5.x branch

My attempt to bisect this failing due to an issue with traitlets getting the wrong type of toolbar (?!)

**Comment by QuLogic** [comment]
Bisect tells me it was fixed by #23113.

**Comment by tacaswell** [comment]
I think this is duplicate of https://github.com/matplotlib/matplotlib/issues/23042 and already fixed.  Sorry for breaking this.

**Comment by tacaswell** [comment]
As a side note, @QuLogic figured out the problem I was having with bisecting:

 - on the v3.5.x branch subversion_scm reports a 3.5.3.dev... version
 - on any other commit teh version is 3.6.0.dev...
 - there is a check in ipympl to do something different on mpl3.6+ which is broken on the 3.5.x line

**Comment by tacaswell** [comment]
And like #23042  adding an explicit `plt.ion()` should fix this on 3.5.2.
